qutebrowser/browser/webengine/interceptor.py

```diff
@@ -85,6 +87,20 @@ def install(self, profile):
         """Install the interceptor on the given QWebEngineProfile."""
         profile.setRequestInterceptor(self)
 
+    # pylint: disable=bad-builtin
+    WHITELIST_REQUEST_METHODS = frozenset(map(QByteArray, [b'GET', b'HEAD']))
+    # pylint: enable=bad-builtin
+    @staticmethod
+    def _handle_redirect(info: QWebEngineUrlRequestInfo, url: QUrl) -> bool:
+        """Handle redirection on qtwebengine."""
+        # Redirecting a request that contains payload data is not allowed.
+        # To be safe, abort on any request not in a whitelist.
+        if (info.requestMethod()
+                not in RequestInterceptor.WHITELIST_REQUEST_METHODS):
+            return False
+        info.redirect(url)
```

Comment:
Maybe it should check if the request has already been marked as blocked or not? You would have to change the redirect method to take the Request too.